oauth2_http/java/com/google/auth/oauth2/MetricsUtils.java

```diff
@@ -67,4 +68,44 @@ private static String getAuthLibraryVersion() {
     }
     return version;
   }
+
+  public enum RequestType {
+    ACCESS_TOKEN_REQUEST("at"),
+    ID_TOKEN_REQUEST("it"),
+    METADATA_SERVER_PIN("mds"),
+    UNSPECIFIED("unspecified");
+
+    private String label;
+
+    private RequestType(String label) {
+      this.label = label;
+    }
+
+    public String getLabel() {
+      return label;
+    }
+  }
+
+  static String getGoogleCredentialsMetricsHeader(
```

Comment:
Could you a small section highlighting the difference between RequestType.UNSPECIFIED, CredentialType.UNKNOWN, and the normal flow?

What happens when both RequestType.UNSPECIFIED and CredentialType.UNKNOWN come in? Or is that not possible? If not, do we want to check that and throw and error?

Reply:
You are right that "both RequestType.UNSPECIFIED and CredentialType.UNKNOWN" is not currently intended for auth metrics. However, I am a bit hesitant to throw error for this utility methods, as it would need signature changes to methods sending the requests (IOException that these requests methods throws is not appropriate).
I am adding a case for this and simply sending language&lib versions. This is not actually used now, but does not hurt either.
Also simplified the logic to make it a bit more clear.

Reply:
We could throw something like a RunTimeException (possibly a IllegalStateException). It's just that it's possible for customers to configure the CredentialType because we have a public setter on the Credentials and currently can't hide it.

I think the worst case is that our metrics may not e 100% accurate if users tinker with it. It probably would be better than crashing the application so I think I'm ok with not throwing anything.